### ISSUE TITLE
Support Azure SQL Data Warehouse

### ISSUE THREAD

**Issue by dbryant4** [hashicorp/terraform#9194]
### Terraform Version
Terraform v0.7.4

### Affected Resource(s)
- azurerm_sql_database

### Terraform Configuration Files
```hcl
provider "azurerm" {
  subscription_id = "..."
  client_id       = "..."
  client_secret   = "..."
  tenant_id       = "..."
}

resource "azurerm_resource_group" "terraform-test" {
   name = "terraform-test"
   location = "West Central US"
}

resource "azurerm_sql_firewall_rule" "terraform-test" {
    name = "FirewallRule1"
    resource_group_name = "${azurerm_resource_group.terraform-test.name}"
    server_name = "${azurerm_sql_server.terraform-test.name}"
    start_ip_address = "0.0.0.0"
    end_ip_address = "255.255.255.255"
}

resource "azurerm_sql_server" "terraform-test" {
    name = "terraform-test-sql-server"
    resource_group_name = "${azurerm_resource_group.terraform-test.name}"
    location = "West Central US"
    version = "12.0"
    administrator_login = "admin"
    administrator_login_password = "SuperSecretPassword123"
}
resource "azurerm_sql_database" "terraform-test" {
    name = "adwtestdb"
    resource_group_name = "${azurerm_resource_group.terraform-test.name}"
    server_name = "${azurerm_sql_server.terraform-test.name}"
    location = "West Central US"
    edition = "DataWarehouse"
    requested_service_objective_name = "DW100"
}

```

### Debug Output
```
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * azurerm_sql_database.terraform-test: SQL Database Edition can only be Basic, Standard or Premium
```

### Panic Output
N/A

### Expected Behavior
Terraform should provision an Azure SQL Data Warehouse

### Actual Behavior
Terraform printed the following error message: `azurerm_sql_database.terraform-test: SQL Database Edition can only be Basic, Standard or Premium`

### Steps to Reproduce
1. `terraform apply` using the above configuration file

### Important Factoids
I believe Azure SQL Data Warehouse is a recent addition to Azure's offerings. 

### References
- [Resource Group template exported from the Azure portal](https://gist.github.com/dbryant4/f0a0afc429f8c6bc1ea72cccdb81eab4). This template covers a simple Azure SQL Data Warehouse instance created using the web portal.

**Comment by stack72** [comment]
Hi @dbryant4 

Thanks for the enhancement request here - I have opened a PR for this. HOpefully this will be included for Terraform 0.7.5 :)

P.

**Comment by dbryant4** [comment]
Awesome! Thanks for the quick turn around! 👍
